test/book/specialized/decker-schmitt-invariant-theory/cox_ring.jlcon

```diff
@@ -9,7 +9,11 @@ julia> G = matrix_group(g1, g2)
 Matrix group of degree 3
   over cyclotomic field of order 3
 
-julia> chi = character_table(G)[6];
+julia> tbl = character_table(G);  F, _ = abelian_closure(QQ);
+
+julia> p = x -> order(kernel(x)[1]) == 8 && x(G(g2)) == F(z_3);
+
+julia> chi = iterate(x for x in tbl if p(x))[1];
```

Comment:
I would find something like
```suggestion
julia> chi = only(filter(p, collect(tbl)));
```
a bit more readable. Probably just a question of taste though.

Reply:
`filter` unfortunately does not work, as char tables are iteratble, but not an AbstractVector. But I just applied something similar to your suggestion

Edit: missed the `collect`. your's should work as well